src/coreclr/src/jit/layout.h

```diff
@@ -195,6 +195,8 @@ class ClassLayout
         }
     }
 
+    static bool AreMatching(const ClassLayout* layout1, const ClassLayout* layout2);
```

Comment:
Nit; I might rename this as something like `AreCompatible`, since it's not really checking for an actual match.